src/annotator/integrations/vitalsource.ts

```diff
@@ -201,21 +206,37 @@ export class VitalSourceContentIntegration
   extends TinyEmitter
   implements Integration
 {
+  private _features: IFeatureFlags;
   private _htmlIntegration: HTMLIntegration;
   private _listeners: ListenerCollection;
   private _textLayer?: ImageTextLayer;
 
-  constructor(container: HTMLElement = document.body) {
+  constructor(
+    /* istanbul ignore next - defaults are overridden in tests */
+    container: HTMLElement = document.body,
+    options: { features: IFeatureFlags }
+  ) {
     super();
 
-    const features = new FeatureFlags();
+    this._features = options.features;
+
+    // If the book_as_single_document flag changed, this will change the
+    // document URI returned by this integration.
+    this._features.on('flagsChanged', () => {
+      this.emit('uriChanged');
+    });
+
+    const htmlFeatures = new FeatureFlags();
 
     // Forcibly enable the side-by-side feature for VS books. This feature is
     // only behind a flag for regular web pages, which are typically more
     // complex and varied than EPUB books.
-    features.update({ html_side_by_side: true });
+    htmlFeatures.update({ html_side_by_side: true });
```

Comment:
I hope the functionality of `FeatureFlags#update` doesn't lead to confusion: it replaces the entirety of the `FeatureFlag` instance's `flags` "state"[^1] — i.e. if a `book_as_single_document` flag had been present on `htmlFeatures`, it would be obliterated by this `update`. I imagine that's by design, but it took me a few minute to unpack the intent here. The `HTMLIntegration` does not have access to any feature flags except for the `html_side_by_side` one here set manually. This feels a bit funky...but I guess the idea is that this `VitalSource` integration "owns' the features for its contained `HTMLIntegration`?

[^1]: Well, kinda: the `flags` reference stays intact but is cleared and re-populated with (only) the flags passed to `update`

Reply:
> I imagine that's by design, but it took me a few minute to unpack the intent here.

I'm actually not certain that the current behavior of replacing feature flags, as opposed to merging them, is the best choice. We could consider changing it if we find it isn't the most useful behavior. In this situation you're right that the VS integration owns the flags it passes to the HTMLIntegration, so there are no inherited flags from elsewhere.